meliae/_scanner_core.c

```diff
@@ -669,14 +670,14 @@ _dump_object_to_ref_info(struct ref_info *info, PyObject *c_obj, int recurse)
         } else if (c_obj == Py_False) {
             _write_static_to_info(info, ", \"value\": \"False\"");
         } else {
-            _write_to_ref_info(info, ", \"value\": %ld", PyLong_AsLong(c_obj));
+            _write_to_ref_info(info, ", \"value\": %lld", PyLong_AsLongLongAndOverflow(c_obj, &overflow));
```

Comment:
You're ignoring overflow here. Any reason to not just call PyLongAsLongLong?

Reply:
I'm ignoring overflow deliberately. It's not possible to serialize Python long as C `long` or `long long` because Python `long` has arbitrary length.

I've simply decided that it's not terribly important to keep precise value here because it's unlikely that in most programs leak would be caused by a long number.